File: exam_app/lib/models/exam_session.dart
import 'package:json_annotation/json_annotation.dart';

part 'exam_session.g.dart';

@JsonSerializable()
class ExamSession {
  final int id;
  final int examId;
  final int userId;
  final SessionStatus status;
  final DateTime? startedAt;
  final DateTime? completedAt;
  final int totalScore;
  final int maxScore;
  final double? percentage;

  ExamSession({
    required this.id,
    required this.examId,
    required this.userId,
    required this.status,
    this.startedAt,
    this.completedAt,
    required this.totalScore,
    required this.maxScore,
    this.percentage,
  });

  factory ExamSession.fromJson(Map<String, dynamic> json) => _$ExamSessionFromJson(json);
  Map<String, dynamic> toJson() => _$ExamSessionToJson(this);
}

enum SessionStatus {
  @JsonValue('STARTED')
  started,
  @JsonValue('IN_PROGRESS')
  inProgress,
  @JsonValue('COMPLETED')
  completed,
  @JsonValue('ABANDONED')
  abandoned,
}

File: exam-api/src/main/java/com/camoleze/examapi/model/ExamSession.java
package com.camoleze.examapi.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("exam_sessions")
public class ExamSession {
    @Id
    private Long id;
    private Long examId;
    private Long userId;
    private SessionStatus status;
    private LocalDateTime startedAt;
    private LocalDateTime completedAt;
    private Integer totalScore;
    private Integer maxScore;
    
    @CreatedDate
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime createdAt;
    
    @LastModifiedDate
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime updatedAt;
    
    public enum SessionStatus {
        STARTED, IN_PROGRESS, COMPLETED, ABANDONED
    }
}